src/main/java/com/mixpanel/android/util/HttpService.java

```diff
@@ -27,8 +27,36 @@
  * An HTTP utility class for internal use in the Mixpanel library. Not thread-safe.
  */
 public class HttpService implements RemoteService {
+
+    private static boolean sAdBlockerEnabled;
+
+    @Override
+    public void checkAdBlockerEnabled() {
+        Thread t = new Thread(new Runnable() {
+            public void run() {
+                try {
+                    InetAddress apiMixpanelInet = InetAddress.getByName("api.mixpanel.com");
+                    InetAddress decideMixpanelInet = InetAddress.getByName("decide.mixpanel.com");
+                    sAdBlockerEnabled = apiMixpanelInet.isLoopbackAddress() ||
+                            apiMixpanelInet.isAnyLocalAddress() ||
+                            decideMixpanelInet.isLoopbackAddress() ||
+                            decideMixpanelInet.isAnyLocalAddress();
+                    if (sAdBlockerEnabled ) {
+                        Log.i(LOGTAG, "AdBlocker is enabled. Won't be able to use Mixpanel services.");
+                    }
+                } catch (UnknownHostException e) {
+                }
+            }
+        });
+
+        t.start();
+    }
+
     @Override
     public boolean isOnline(Context context) {
+        if (sAdBlockerEnabled)
```

Comment:
This is picky but can you add `{}` to wrap this block, just to match how we do it in other places?

eg.

```
            if (null == bitmap) {
                throw new CantGetImageException("Downloaded data could not be interpreted as a bitmap");
            }
```